raiden/network/proxies/token_network.py

```diff
@@ -0,0 +1,773 @@
+        # partner address.
+        if other_peer not in self.open_channel_transactions:
+            new_open_channel_transaction = AsyncResult()
+            self.open_channel_transactions[other_peer] = new_open_channel_transaction
+
+            try:
+                transaction_hash = self._new_netting_channel(other_peer, settle_timeout)
+            except Exception as e:
+                new_open_channel_transaction.set_exception(e)
+                raise
+            else:
+                new_open_channel_transaction.set(transaction_hash)
+            finally:
+                self.open_channel_transactions.pop(other_peer, None)
+        else:
+            # All other concurrent threads should block on the result of opening this channel
+            transaction_hash = self.open_channel_transactions[other_peer].get()
+
+        netting_channel_results_encoded = self.proxy.call(
+            'getChannelInfo',
+            self.node_address,
+            other_peer,
+        )
+
+        # channel identifier is at index 0
+        channel_identifier = netting_channel_results_encoded[0]
+
+        log.debug('channel_identifier {}'.format(channel_identifier))
+
+        if not channel_identifier:
+            log.error(
+                'channel_identifier failed',
+                peer1=pex(self.node_address),
+                peer2=pex(other_peer)
+            )
+            raise RuntimeError('channel_identifier failed')
+
+        log.info(
+            'new_netting_channel called',
+            peer1=pex(self.node_address),
+            peer2=pex(other_peer),
+            channel_identifier=channel_identifier,
+        )
+
+        return channel_identifier
+
+    def _new_netting_channel(self, other_peer, settle_timeout):
+        if self.channel_exists(other_peer):
+            raise DuplicatedChannelError('Channel with given partner address already exists')
+
+        transaction_hash = estimate_and_transact(
+            self.proxy,
+            'openChannel',
+            self.node_address,
+            other_peer,
+            settle_timeout,
+        )
+
+        if not transaction_hash:
+            raise RuntimeError('open channel transaction failed')
+
+        self.client.poll(unhexlify(transaction_hash), timeout=self.poll_timeout)
+
+        if check_transaction_threw(self.client, transaction_hash):
+            raise DuplicatedChannelError('Duplicated channel')
+
+        return transaction_hash
+
+    def channel_exists(self, partner: typing.Address) -> bool:
+        channel_data = self.detail_channel(partner)
+
+        if not isinstance(channel_data['state'], typing.T_ChannelState):
+            raise ValueError('channel state must be of type ChannelState')
+
+        log.debug('existing_channel {}'.format(channel_data))
+
+        return channel_data['state'] > CHANNEL_STATE_NONEXISTENT
+
+    def _detail_participant(self, participant: typing.Address, partner: typing.Address):
```

Comment:
Why do you make this "private" with the `_` prefix while the next function, `detail_channel` and `detail_participant` is not?